### ISSUE TITLE
mc --force does not work for this case

### ISSUE THREAD

**Issue by krishnasrinivas** [minio/mc#1432]
```
krishna@escape:~/gocode/src/github.com/minio$ mc rm -r s3/peppadog/pkg --force
mc: <ERROR> Recursive removal requires --force option. Please review carefully before performing this *DANGEROUS* operation. 
```

i.e if s3/peppadog/pkg is between -r and --force

**Comment by harshavardhana** [comment]
Works with 

```
mc rm --recursive --force s3/bs-blobs/usr/share
```

Looks like this requires cli change, its a cli related bug.

**Comment by harshavardhana** [comment]
Tested this and tried fixing the CLI package. The bug is how the indexing is handled in CLI for incoming cli arguments. 

The problem comes because this is how it is being parsed as 

```
--recursive --force <bucket>
```

```
<bucket> --recursive --force
```

Both these work for the simple reason that it is easier for the code to parse arguments if they are at known positions. 

It is really hard when we split the flags

```
--recursive <bucket> --force
```

```
--force <bucket> --recursive
``` 

Both of them do not work, i tried simplifying the cli code to handle this but there are many cases where this is a problem. 

For example like this 

```
--content-type image/png
--content-type=image/png 
``` 

Both are supported and following style will work 

```
--content-type image/png <bucket>
<bucket> --content-type image/png will work
``` 

But the styles like below, clobbers the parser and leads to bugs.  

```
--content-type image/png <bucket> --recursive
``` 

The right fix is to enhance the cli package to have very clear distinction on how '--flags' take their input arguments.  Should we enforce '=' or should we not is something which is left for future. 

For the time being we have made sure to document flags either at the beginning of the command or at the end.
